src/plugins/intel_npu/src/common/CMakeLists.txt

```diff
@@ -0,0 +1,40 @@
+# Copyright (C) 2018-2024 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+#
+
+set(TARGET_NAME openvino_npu_common)
```

Comment:
Instead of creating a new target, could we consider merging the compiler and backend target into the plugin target? Not seeing much benefit of separating these.

It actually slows down the build by coping the same object files into a lib archive before actual linking of plugin.

For tests we could export some interface target just for the headers from the plugin cmake or use explicit include path there.

Reply:
Agree. Even common could be merged into one.